neuralmonkey/decoders/autoregressive.py

```diff
@@ -421,10 +433,87 @@ def loop_continue_criterion(self, *args) -> tf.Tensor:
                                  self.max_output_len)
         return tf.logical_and(not_all_done, before_max_len)
 
+    def next_state(self, loop_state: LoopState) -> Tuple[tf.Tensor, Any, Any]:
+        raise NotImplementedError("Abstract method.")
+
     def get_body(self, train_mode: bool, sample: bool = False,
-                 temperature: float = 1) -> Callable:
+                 temperature: float = 1.) -> Callable:
         """Return the while loop body function."""
-        raise NotImplementedError("Abstract method")
+
+        def is_finished(finished: tf.Tensor, symbols: tf.Tensor) -> tf.Tensor:
+            has_just_finished = tf.equal(symbols, END_TOKEN_INDEX)
+            return tf.logical_or(finished, has_just_finished)
+
+        def state_to_logits(state: tf.Tensor) -> tf.Tensor:
+            logits = tf.matmul(state, self.decoding_w)
+            logits += self.decoding_b
+
+            if self.supress_unk:
+                unk_mask = tf.one_hot(
+                    UNK_TOKEN_INDEX, depth=len(self.vocabulary), on_value=-1e0)
```

Comment:
Ta `on_value` by měla být řádově míň, ne?

Reply:
Jo, tohle je preklep. Diky, zes ho nasel (supress_unk nepouzivam, takze jsem to netestoval).